sdcm/nemesis.py

```diff
@@ -440,41 +440,47 @@ def _terminate_cluster_node(self, node):
         self.cluster.terminate_node(node)
         self.monitoring_set.reconfigure_scylla_monitoring()
 
-    def disrupt_nodetool_decommission(self, add_node=True):
-        # pylint: disable=too-many-locals
-
-        def get_node_info_list(verification_node):
+    def _decommission_cluster_node(self, node):
+        def get_node_ip_list(verification_node):
             try:
-                return self.cluster.get_node_info_list(verification_node)
+                ip_node_list = []
+                status = self.cluster.get_nodetool_status(verification_node)
+                for nodes_ips in status.values():
+                    ip_node_list.extend(nodes_ips.keys())
+                return ip_node_list
             except Exception as details:  # pylint: disable=broad-except
                 self.log.error(str(details))
                 return None
 
-        self._set_current_disruption('Decommission %s' % self.target_node)
-        target_node_ip = self.target_node.ip_address
-        self.target_node.run_nodetool("decommission")
+        target_node_ip = node.ip_address
+        node.run_nodetool("decommission")
         verification_node = random.choice(self.cluster.nodes)
-        node_info_list = get_node_info_list(verification_node)
-        while verification_node == self.target_node or node_info_list is None:
+        node_ip_list = get_node_ip_list(verification_node)
+        while verification_node == node or node_ip_list is None:
             verification_node = random.choice(self.cluster.nodes)
-            node_info_list = get_node_info_list(verification_node)
-
-        nodetool_ips = [node_info['ip'] for node_info in node_info_list]
-        error_msg = ('Node that was decommissioned %s still in the cluster. '
-                     'Cluster status info: %s' % (self.target_node,
-                                                  node_info_list))
-        if target_node_ip in nodetool_ips:
-            self.log.info('Decommission %s FAIL', self.target_node)
+            node_ip_list = get_node_ip_list(verification_node)
+
+        if target_node_ip in node_ip_list:
+            cluster_status = self.cluster.get_nodetool_status(verification_node)
+            error_msg = ('Node that was decommissioned %s still in the cluster. '
+                         'Cluster status info: %s' % (node,
+                                                      cluster_status))
+
+            self.log.info('Decommission %s FAIL', node)
```

Comment:
- raise an exception to fail the Nemesis, otherwise, its status will be OK in the email report
- when the exception is thrown no need to return result

Reply:
@bentsi , Added new Exception and raise it if decommission command finished, but node stay in cluster.